agent/exec/controller.go

```diff
@@ -288,7 +288,7 @@ func Do(ctx context.Context, task *api.Task, ctlr Controller) (*api.TaskStatus,
 		status.PortStatus = portStatus
 	}()
 
-	if task.DesiredState == api.TaskStateShutdown {
+	if task.DesiredState == api.TaskStateShutdown || task.DesiredState == api.TaskStateRemove {
```

Comment:
why not `if task.DesiredState >= api.TaskStateShutdown`, to encompass possible future desired states?

Reply:
I don't have a solid answer for this yet, I believe I did the exact comparison since the previous condition was an exact comparison itself.

Reply:
I think you can go with `>= api.TaskStateShutdown` here. This branch bounds the largest state achievable in the agent as `SHUTDOWN`, which is the behavior we want.